### PR TITLE
[[ ObjectGC ]] Make the engine delete objects more frequently

### PR DESCRIPTION
**Pull request by runrevmark**
The engine will now try to delete objects as soon as it can, rather than deferring the deletion until the main event loop is returned to.

This means that tight loops which create and delete objects should no longer suffer from memory overflow assuming they aren't being created and deleted in different (sibling) wait loops.

### CODE REVIEW

File: engine/src/object.cpp [hunk 1]
```diff
@@ -4763,3 +4772,230 @@ bool MCObjectHandle::Exists(void)
...
+struct MCDeletedObjectPool
+{
+    uindex_t references;
+    MCDeletedObjectPool *parent;
+    MCObject *to_delete;
+    bool defunct : 1;
+};
+
+MCDeletedObjectPool *MCsparedeletedobjectpool = nil;
+MCDeletedObjectPool *MCdeletedobjectpool = nil;
```
Comment (livecodestephen) [inline engine/src/object.cpp]: Could these be static variables?

**Comment by peter-b** [comment]
@runrevmark: I like the `MCActionsRunAll()` abstraction! Does it make sense to have more abstract `MCWaitRunEnterHook()` and `MCWaitRunExitHook()` functions too (not sure about the naming)?

**Comment by runrevmark** [comment]
@peter-b: Perhaps in the future if there's something else we add that needs to do a similar thing - it would be easy enough to refactor at that point, probably better doing that rather than 'pre-guessing' what we might need.

File: engine/src/redraw.cpp [hunk 2]
```diff
@@ -1233,14 +1232,16 @@ void MCRedrawRestoreLockScreen(uint2 p_lock)
 {
 	MClockscreen = p_lock;
 	
-	MCredrawupdatescreenneeded = MClockscreen == 0 && s_screen_is_dirty && !s_screen_updates_disabled;
+	if (MClockscreen == 0 && s_screen_is_dirty && !s_screen_updates_disabled)
+        MCActionsSchedule(kMCActionsUpdateScreen);
```
Comment (peter-b) [inline engine/src/redraw.cpp:1236]: Since this particular test & call is run repeatedly from multiple places, it may make sense to factor it out.

**Comment by peter-b** [comment]
Nothing is jumping out at me as a problem here. :+1: 

@livecode-vulcan review ok 2c542fb

**Comment by livecode-vulcan** [comment]
💙 review by @peter-b ok 2c542fbc0ea40dbdc95b54e4d6b1789e7552eb5a

**Comment by livecode-vulcan** [comment]
😞 test failure 2c542fbc0ea40dbdc95b54e4d6b1789e7552eb5a
- try-linux-x86: [failure](https://vulcan.livecode.com/builders/try-linux-x86/builds/207)
- try-win-x86: [failure](https://vulcan.livecode.com/builders/try-win-x86/builds/271)
- try-ios: [pending](https://vulcan.livecode.com/builders/try-ios/builds/160)
- try-android-armv6: [failure](https://vulcan.livecode.com/builders/try-android-armv6/builds/218)
- try-linux-x86_64: [failure](https://vulcan.livecode.com/builders/try-linux-x86_64/builds/153)
- try-mac: unknown

**Comment by peter-b** [comment]
@livecode-vulcan review ok 0f1197d

**Comment by livecode-vulcan** [comment]
💙 review by @peter-b ok 0f1197d222bf54958e119eb5f677dad81f1f47bf

**Comment by livecode-vulcan** [comment]
😎 test success 0f1197d222bf54958e119eb5f677dad81f1f47bf
- try-linux-x86: [success](https://vulcan.livecode.com/builders/try-linux-x86/builds/401)
- try-win-x86: [success](https://vulcan.livecode.com/builders/try-win-x86/builds/463)
- try-ios: [success](https://vulcan.livecode.com/builders/try-ios/builds/347)
- try-android-armv6: [success](https://vulcan.livecode.com/builders/try-android-armv6/builds/412)
- try-linux-x86_64: [success](https://vulcan.livecode.com/builders/try-linux-x86_64/builds/347)
- try-mac: [success](https://vulcan.livecode.com/builders/try-mac/builds/338)

**Pull request merged**